File: python/all-your-base/all_your_base.py
def rebase(input_base, digits, output_base):
  if input_base <= 1 or output_base <= 1:
    raise ValueError('invalid')

  base_10_num = 0

  for i, digit in enumerate(reversed(digits)):
    if digit < 0 or digit >= input_base:
      raise ValueError('invalid')
    base_10_num += digit * (input_base ** i)
  
  res = []

  while base_10_num > 0:
    res = [base_10_num % output_base] + res
    base_10_num //= output_base

  return res

File: csharp/all-your-base/AllYourBase.cs
using System;
using System.Collections.Generic;

public static class AllYourBase
{
    public static int[] Rebase(int inputBase, int[] inputDigits, int outputBase)
    {
        int base10 = 0;
        var outputDigits = new List<int>();

        if (inputBase < 2 || outputBase < 2) {
            throw new ArgumentException("Invalid base.");
        }

        int exp = inputDigits.Length - 1;
        foreach (var digit in inputDigits) {
            if (digit < 0 || digit >= inputBase) {
                throw new ArgumentException("Invalid digit.");
            }
            base10 += digit * (int)Math.Pow(inputBase, exp--);
        }

        if (base10 == 0) {
            return new []{ 0 };
        }

        while (base10 > 0) {
            outputDigits.Insert(0, base10 % outputBase);
            base10 /= outputBase;
        }

        return outputDigits.ToArray();
    }
}